aws-distro-opentelemetry-node-autoinstrumentation/src/patches/aws/services/bedrock.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { Attributes, DiagLogger, Span, SpanKind, Tracer } from '@opentelemetry/api';
+import {
+  AwsSdkInstrumentationConfig,
+  NormalizedRequest,
+  NormalizedResponse,
+} from '@opentelemetry/instrumentation-aws-sdk';
+import { AWS_ATTRIBUTE_KEYS } from '../../../aws-attribute-keys';
+import { RequestMetadata, ServiceExtension } from '../../../third-party/otel/aws/services/ServiceExtension';
+import { AwsSpanProcessingUtil } from '../../../aws-span-processing-util';
+
+const AGENT_ID: string = 'agentId';
+const KNOWLEDGE_BASE_ID: string = 'knowledgeBaseId';
+const DATA_SOURCE_ID: string = 'dataSourceId';
+const GUARDRAIL_ID: string = 'guardrailId';
+const MODEL_ID: string = 'modelId';
+const AWS_BEDROCK_SYSTEM: string = 'aws_bedrock';
+
+const AGENT_OPERATIONS = [
+  'CreateAgentActionGroup',
+  'CreateAgentAlias',
+  'DeleteAgentActionGroup',
+  'DeleteAgentAlias',
+  'DeleteAgent',
+  'DeleteAgentVersion',
+  'GetAgentActionGroup',
+  'GetAgentAlias',
+  'GetAgent',
+  'GetAgentVersion',
+  'ListAgentActionGroups',
+  'ListAgentAliases',
+  'ListAgentKnowledgeBases',
+  'ListAgentVersions',
+  'PrepareAgent',
+  'UpdateAgentActionGroup',
+  'UpdateAgentAlias',
+  'UpdateAgent',
+];
+
+const KNOWLEDGE_BASE_OPERATIONS = [
+  'AssociateAgentKnowledgeBase',
+  'CreateDataSource',
+  'DeleteKnowledgeBase',
+  'DisassociateAgentKnowledgeBase',
+  'GetAgentKnowledgeBase',
+  'GetKnowledgeBase',
+  'ListDataSources',
+  'UpdateAgentKnowledgeBase',
+];
+
+const DATA_SOURCE_OPERATIONS = ['DeleteDataSource', 'GetDataSource', 'UpdateDataSource'];
+
+const operationToBedrockAgentServiceMapping: { [key: string]: { [key: string]: string } } = {};
+for (const operation of AGENT_OPERATIONS) {
+  operationToBedrockAgentServiceMapping[operation] = { [AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_AGENT_ID]: AGENT_ID };
+}
+for (const operation of KNOWLEDGE_BASE_OPERATIONS) {
+  operationToBedrockAgentServiceMapping[operation] = {
+    [AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_KNOWLEDGE_BASE_ID]: KNOWLEDGE_BASE_ID,
```

Comment:
Confirmed with Zhonghao that logically there is no reason to split the two. Request and response attributes exist in the request and response objects respectively, and so dumping them into a single object is not a problem since we can loop through them and just check if they exist in the request or response to find any that exist.